p1-staking-pool-dyn/src/vault.rs

```diff
@@ -26,10 +26,16 @@ impl Vault {
 }
 
 impl Contract {
+    #[inline]
     pub(crate) fn get_vault_or_default(&self, account_id: &AccountId) -> Vault {
         self.vaults.get(account_id).unwrap_or_default()
     }
 
+    #[inline]
```

Comment:
remove this.
this plus the change you made above will not work.
leave the old function: `get_vault_or_default`

Reply:
Because we can't use `get_vault_or_default` everywhere. See my comment above. Both `close` and `withdraw_crop` should fail if a user didn't deposit / stake anything.